ui/src/variables/utils/hydrateVars.ts

```diff
@@ -126,9 +126,10 @@ const findSubgraph = (
   variables: Variable[]
 ): VariableNode[] => {
   const subgraph: Set<VariableNode> = new Set()
-
   // use an ID array to reduce the chance of reference errors
   const varIDs = variables.map(v => v.id)
+  // create an array of IDs to reference later
+  const graphIDs = []
```

Comment:
note for self: this was because `Set.prototype.has` compares by reference, which we dont guarantee in an immutable kind of world